bin/man-n

```diff
@@ -22,38 +22,45 @@ cat << EOF
   Options:
 
     -h, --help  output usage information
-    -l, --link  link \`man-n\` to \`man n\`
+    -l, --link  default to man(1)
 
   Examples:
 
-    $ man-n browserify  # show the browserify documentation
-    $ man-n --link      # link \`man-n\` to \`man n\`
-    $ man n browserify  # show the browserify documentation
+    $ man-n browserify          # show browserify docs
+    $ man-n --link ls           # show ls(1) docs
+    $ man-n --link n browserify # show browserify docs
 EOF
 }
 
 #
-# If `--linked`, detect if this is a `n` page.
+# Switch between man(1) and man-n.
 #
 
-if [ "$1" = "--linked" ]; then shift;
-  if [ "$1" = "n" ]; then shift;
+link () {
+  shift
+  if [ "$1" = "n" ]; then
+    shift
+    switch "$@"
   else
     "$original" "$@"
     exit
   fi
-fi
+}
 
 #
-# CLI flags
+# CLI flags.
 #
 
-case $1 in
-  "")         usage; exit 1 ;;
-  -h|--help)  usage; exit ;;
-  --link)     echo 'alias "man=man-n --linked"'; exit ;;
-  *)          readonly name=$1 ;;
-esac
+switch () {
+  case $1 in
+    "")         usage; exit 1 ;;
+    -h|--help)  usage; exit ;;
+    -l|--link)  link "$@" ;;
+    *)          readonly name=$1 ;;
+  esac
+}
+
+switch "$@"
```

Comment:
Small note for simplicity's sake: rather than calling `switch` at this point, you could also simply call the original `$0` with `$@` at that point. This would move the recursion to the top of the script, making it perhaps simpler to reason about.